pulpcore/app/util.py

```diff
@@ -55,6 +56,45 @@ def get_url(model, domain=None):
     return reverse(get_view_name_for_model(model, view_action), kwargs=kwargs)
 
 
+def get_prn(instance=None, uri=None):
+    """
+    Get a Pulp Resource Name (PRN) for the specified model instance. It is similar to a HREF
+    url in that it uniquely identifies a resource, but it also has the guarantee that it will not
+    change regardless of API_ROOT or DOMAIN_ENABLED. This is used in our resource locking/
+    reservation code to identify resources.
+
+    The format for the PRN is as follows:
+    ```
+        prn:model-label-lower:id
```

Comment:
```suggestion
        prn:model-label-lower:pk
```

BTW: This makes `"shared:prn:label:pk"` for readonly locks. Probably not an issue.